docs/guides/tutorials-appium/appium-cross-platform-automation.md

```diff
@@ -0,0 +1,159 @@
+
+<h3 id="test_setup">Test setup</h3>
+Cross-platform Test Setup must be different from the normal setup, because the process of selecting a platform  must be dynamic. This is a good example for a cross-platform setup:
+
+{% highlight java %}
+public class AbstractTest {
+
+    @Rule
+    public TestObjectTestResultWatcher resultWatcher = new TestObjectTestResultWatcher();
+
+    private AppiumDriver driver;
+    protected Komoot app;
+
+    @Before
+    public void connect() throws Exception {
+        this.driver = new AppiumDriverBuilder()
+                .withTestResultWatcher(resultWatcher)
+                .build();
+
+        resultWatcher.setAppiumDriver(driver);
+
+        app = new Komoot(driver);
+    }
+}
+
+{% endhighlight %}
+
+And the Appium Driver Builder will be:
+
+{% highlight java %}
+public class AppiumDriverBuilder {
+    // To resgister your tests results on TestObject
+    private TestObjectTestResultWatcher resultWatcher;
+
+    public AppiumDriverBuilder withTestResultWatcher(TestObjectTestResultWatcher resultWatcher) {
+        this.resultWatcher = resultWatcher;
+        return this;
+    }
+
+    public AppiumDriver build() throws Exception {
+        // Some Desired Capabilities
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+
+        capabilities.setCapability("testobject_api_key", resultWatcher.getApiKey());
+        capabilities.setCapability("testobject_test_report_id", resultWatcher.getTestReportId());
+
+        capabilities.setCapability("deviceName", "testDevice");
+
+        // Get the platform ("android" or "ios") we use from the environment variables.
+        String platform = System.getenv("PLATFORM");
+
+        // Here we determine if the Test will be running on Android or on iOS
+        if (platform.equalsIgnoreCase("ios")) {
+            capabilities.setCapability("autoDismissAlerts", true);
+            return new IOSDriver(resultWatcher.getTestObjectOrLocalAppiumEndpointURL(), capabilities);
+        } else if (platform.equalsIgnoreCase("android")) {
+            return new AndroidDriver(resultWatcher.getTestObjectOrLocalAppiumEndpointURL(), capabilities);
+        } else {
+            throw new Exception("Unable to read device platform.");
+        }
+    }
+}
+{% endhighlight %}
+
+AppiumDriverBuilder is basically a helper class that helps you to connect to TestObject. So you have TestObject Test Result Watcher which sending your results to TestObject and some other capabilities to establish the connection, also you have the  "if else" statement which determines on which platform ("android" or "ios") you are going to run your test.
```

Comment:
Also, after this we can add the example for using device descriptors instead of environment variables: like we did on AldiTalk.